test/gtest/lattices.cpp

```diff
@@ -546,3 +547,79 @@ TEST(ValTypeLattice, Meet) {
                   Type(HeapType::array, Nullable),
                   Type(HeapType::eq, Nullable));
 }
+
+TEST(SharedLattice, GetBottom) {
+  analysis::Shared<analysis::UInt32> shared{analysis::UInt32{}};
+  EXPECT_EQ(*shared.getBottom(), 0u);
+}
+
+TEST(SharedLattice, Compare) {
+  analysis::Shared<analysis::UInt32> shared{analysis::UInt32{}};
+
+  auto zero = shared.getBottom();
+
+  auto one = zero;
```

Comment:
I am having trouble reading a line that says literally `1 == 0`...

(And later down I see `2 == 1`. By induction, all natural numbers are apparently `0`... :thinking: :laughing: )

How about `a, b, c`?

Reply:
The only way to increase a shared value is via a `join`, so to create our `one` value we have to start from zero and add one via the join on the next line. I'll clarify this in the code.